game.py

```diff
@@ -174,16 +177,12 @@ def _all_clear(self):
             your_time_shape.hide()
         for highscore_shape in self._best_time:
             highscore_shape.hide()
-        for dot in self._dots:
-            if dot.type:
-                dot.type = False
-                dot.set_shape(self._new_dot(self._colors[FILL],
-                                            self._dot_size))
-            dot.set_label('')
-            dot.set_layer(100)
+        for sprite in self._dots:
+            sprite.hide()
+        self._init_grid()
         self._turtle.set_layer(100)
         # Recenter the turtle
-        self._move_turtle(self._dots[int(THIRTEEN * THIRTEEN / 2)].get_xy())
+        self._move_turtle(self._dots[int(self.size * (self.size + (0 if self.size % 2 else 1)) / 2)].get_xy())
```

Comment:
Does this look good?
```suggestion
        self._move_turtle(self._dots[int(self.size * (self.size + 1 - self.size % 2) / 2)].get_xy())
```

Reply:
That's Perfect, thanks for the suggestion!